daemon/algod/api/server/v2/handlers.go

```diff
@@ -463,3 +463,35 @@ func (v2 *Handlers) GetPendingTransactions(ctx echo.Context, params generated.Ge
 func (v2 *Handlers) GetPendingTransactionsByAddress(ctx echo.Context, addr string, params generated.GetPendingTransactionsByAddressParams) error {
 	return v2.getPendingTransactions(ctx, params.Max, params.Format, &addr)
 }
+
+// GetApplicationByID returns application information by app idx.
+// (GET /v2/applications/{application-id})
+func (v2 *Handlers) GetApplicationByID(ctx echo.Context, applicationID uint64) error {
+	appIdx := basics.AppIndex(applicationID)
```

Comment:
I can't find the function for some reason but can't it fail? what happens in case of a non-existing ID?

Reply:
this line is casting from uint64 to basics.AppIndex